xgcm/grid.py

```diff
@@ -1359,6 +1361,26 @@ def iterate_axis_combinations(items):
         # return the product of the metrics
         return functools.reduce(operator.mul, metric_vars, 1)
 
+    def interp_like(self, da_source, metric):
```

Comment:
I find the naming a bit confusing. `da_source` to me implies 'the dataarray that will be interpolated' onto a 'target'.
I would replace in the following way:
`da_source --> da_target`
`metric --> da_source`

The reason why I would not use anything 'metric' here is because this function has much broader use cases which do not necessarily involve metrics.